### PR TITLE
LPS-101288 showing field name when labels are repeated

### PR DESCRIPTION
**Pull request by linolaoj**
This fix implements the behavior discussed with @natocesarrego.
https://issues.liferay.com/browse/LPS-101288
cc: @inacionery
Thanks!

### CODE REVIEW

**Comment by linolaoj** [comment]
ci:test:sf

**Comment by linolaoj** [comment]
ci:test:relevant

**Comment by linolaoj** [comment]
ci:test:relevant

**Comment by inacionery** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by natocesarrego** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/test/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordCSVWriterTest.java [hunk 1]
```diff
@@ -85,8 +85,8 @@ public void testWrite() throws Exception {
 		StringBundler sb = new StringBundler(3);
 
 		sb.append("Field 1,Field 2,Field 3,Field 4\n");
-		sb.append("2,esta é uma 'string',false,11.7\n");
-		sb.append("1,esta é uma 'string',,10");
+		sb.append("2,false,esta é uma 'string',11.7\n");
+		sb.append("1,,esta é uma 'string',10");
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/test/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordCSVWriterTest.java]: Hi @linolaoj,

Can you please explain why these changes were necessary?

Thanks.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/test/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordCSVWriterTest.java [hunk 2]
```diff
@@ -96,118 +96,44 @@ public void testWrite() throws Exception {
 	}
 
 	@Test
-	public void testWriteAfterChangeFieldName() throws Exception {
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/test/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordCSVWriterTest.java]: Hey @linolaoj,

Why did you deleted this test instead of change its expected result?

Thanks.

Comment (linolaoj) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/test/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordCSVWriterTest.java]: Hi @natocesarrego,
this was accidentally removed on the revert, I'll add it back and update according with the decided behavior.
Thanks!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/test/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordCSVWriterTest.java (hunk 1 above)
Comment (linolaoj) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/test/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordCSVWriterTest.java]: @natocesarrego, this was reverted since to the way it was before LPS-101288, since the decided behavior change.

**Comment by linolaoj** [comment]
new pull https://github.com/natocesarrego/liferay-portal/pull/180

**Comment by natocesarrego** [comment]
Thanks for your answers, @linolaoj!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/test/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordCSVWriterTest.java (hunk 1 above)
Comment (natocesarrego) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/test/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordCSVWriterTest.java]: Hi @linolaoj,

Analyzing your explanation more carefully, I realized that this modification is actually setting the value of `field2` to the `field3` and vice-versa, is it intended? I understand that this change only makes sense if the labels were changed too, like ...,`Field3`, `Field2`, etc. Even though, I didn't see how this change is related to the fix. 😞

Comment (linolaoj) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/test/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordCSVWriterTest.java]: Hi @natocesarrego, this was the revert of b0ce331adc5018e7e48adf44f0fc4eb1ed5790ac. This test was updated for the behavior from e1f4091d2070b5d8d6a0698fc548fe0018680f21. And I reverted all the changes of LPS-101288 since the expected behavior changed.
